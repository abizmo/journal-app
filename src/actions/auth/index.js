export { default as login } from './login';
export { default as loginWithEmail } from './loginWithEmail';
export { default as loginWithGoogle } from './loginWithGoogle';
export { default as logout } from './logout';
export { default as signOut } from './signOut';
export { default as signUp } from './signUp';

export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export const types = {
  LOGIN,
  LOGOUT,
};
