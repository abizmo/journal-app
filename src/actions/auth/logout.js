import { LOGOUT } from ".";

const logout = () => ({
  type: LOGOUT,
});

export default logout;
