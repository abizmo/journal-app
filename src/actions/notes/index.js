export { default as setActive } from './setActive';
export { default as refreshNote } from './refreshNote';
export { default as clearNotes } from './clearNotes';
export { default as getNotes } from './getNotes';
export { default as uploadImage } from './uploadImage';
export { default as removeNote } from './removeNote';
export { default as newNote } from './newNote';
export { default as saveNote } from './saveNote';

export const ADD_NOTE = 'ADD_NOTE';
export const CLEAR_NOTES = 'CLEAR_NOTES'
export const DELETE_NOTE = 'DELETE_NOTE';
export const SET_ACTIVE = 'SET_ACTIVE';
export const SET_NOTES = 'SET_NOTES';
export const UPDATE_NOTE = 'UPDATE_NOTE';

export const types = {
  ADD_NOTE,
  CLEAR_NOTES,
  DELETE_NOTE,
  SET_ACTIVE,
  SET_NOTES,
  UPDATE_NOTE,
};
