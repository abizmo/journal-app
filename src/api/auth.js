import {
  createUserWithEmailAndPassword,
  onAuthStateChanged,
  signInWithEmailAndPassword,
  signInWithPopup,
  signOut as firebaseSignOut,
  updateProfile,
} from "firebase/auth";

import { auth, provider } from "../firebase";

export const onCurrentUserChanged = (cb) => onAuthStateChanged(auth, cb);
  
export const signIn = async ({ email, password }) => {
  const { user } = await signInWithEmailAndPassword(auth, email, password)

  return {
    name: user.displayName,
    userId: user.uid
  };
};

export const signInWithGoogle = () => signInWithPopup(auth, provider)
  .then(({ user }) => ({
    name: user.displayName,
    userId: user.uid
}));

export const signOut = () => firebaseSignOut(auth);

export const signUpWithEmailAndPassword = ({
  email,
  name,
  password
}) => createUserWithEmailAndPassword(auth, email, password)
  .then(({ user }) => updateProfile(user, { displayName: name })
  .then(() => ({ name, userId: user.uid})));
