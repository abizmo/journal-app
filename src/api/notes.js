import {
  addDoc,
  collection,
  doc,
  getDocs,
  updateDoc,
} from "firebase/firestore";
import { db } from "../firebase";

export const createNote = (userId, note) => addDoc(collection(db, `${userId}/journal/notes/`), note)
  .then(({ id }) => id);

export const getAllNotes = (userId) => getDocs(collection(db, `${userId}/journal/notes/`))
  .then((res) => {
    const notes = [];
    res.forEach((doc) => {
      notes.push({ ...doc.data(), id: doc.id });
    });
    
    return notes;
});

export const updateNote = (userId, noteId, data) => {
  const docRef = doc(db, `${userId}/journal/notes/${noteId}`);

  return updateDoc(docRef, data)
    .then(() => data);
};