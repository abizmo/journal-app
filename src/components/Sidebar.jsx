import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { signOut } from '../actions/authActions';
import { newNote } from '../actions/notes';
import JournalEntries from './JournalEntries';

const Sidebar = () => {
  const dispatch = useDispatch();
  const { name } = useSelector(state => state.auth)

  const handleNewNote = () => {
    dispatch(newNote());
  };

  const handleLogout = () => {
    dispatch(signOut());
  };

  return (
    <aside className="journal__sidebar-container">
      <div className="journal__sidebar-nav">
        <b>
          <i className="far fa-heart"></i>
          <span> { name }</span>
        </b>
        <button
          className="btn btn-outline-white"
          onClick={ handleLogout }
        >Logout</button>
      </div>
      <div
        className="journal__sidebar-new"
        onClick={ handleNewNote }
      >
        <i className="far fa-plus-square fa-5x"></i>
        <p className="mt-2">New entry</p>
      </div>
      <JournalEntries />
    </aside>
  );
};

export default Sidebar;
