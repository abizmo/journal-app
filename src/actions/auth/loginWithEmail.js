import { login } from ".";

import { startAction, stopAction } from "../ui";
import { signIn } from "../../api/auth";
import { error } from "../../helpers/alert";

const loginWithEmail = ({ email, password }) => async (dispatch) => {
  try {
    dispatch(startAction());
    const user = await signIn({ email, password });
    dispatch(login(user));
  } catch ({ message }) {
    error(message);
  }
  dispatch(stopAction());
};

export default loginWithEmail;
