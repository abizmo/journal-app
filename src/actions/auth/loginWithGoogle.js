import { login } from ".";

import { signInWithGoogle } from "../../api/auth";
import { error } from "../../helpers/alert";

const loginWithGoogle = () => async (dispatch) => {
  try {
    const user = await signInWithGoogle();
    dispatch(login(user));
  } catch ({ message }) {
    error(message);
  }
};

export default loginWithGoogle;
