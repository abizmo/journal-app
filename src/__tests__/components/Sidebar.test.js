import { mount } from "enzyme";
import { Provider } from "react-redux";
import createMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import { signOut } from "../../actions/auth";
import { newNote } from "../../actions/notes";
import Sidebar from '../../components/Sidebar';
jest.mock('../../actions/auth');
jest.mock('../../actions/notes');

const initialState = {
  auth: {
    name: 'goccita',
  },
  notes: {
    notes: [],
  },
};
const middlewares = [thunk];
const mockStore = createMockStore(middlewares);

describe('<Sidebar />', () => {
  const store = mockStore(initialState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <Sidebar />
    </Provider>
  );

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text().trim()).toBe('goccita');
  });
  
  test('should dispatch signOut', () => {
    wrapper.find('button').simulate('click');

    expect(signOut).toHaveBeenCalled();
  });
  
  test('should dispatch newNote', () => {
    wrapper.find('#new').simulate('click');
    
    expect(newNote).toHaveBeenCalled();
  });
});
