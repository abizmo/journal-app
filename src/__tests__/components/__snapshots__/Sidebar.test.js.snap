// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render and match snapshot 1`] = `
<aside
  className="journal__sidebar-container"
>
  <div
    className="journal__sidebar-nav"
  >
    <b>
      <i
        className="far fa-heart"
      />
      <span>
         
        goccita
      </span>
    </b>
    <button
      className="btn btn-outline-white"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__sidebar-new"
    id="new"
    onClick={[Function]}
  >
    <i
      className="far fa-plus-square fa-5x"
    />
    <p
      className="mt-2"
    >
      New entry
    </p>
  </div>
  <div
    className="journal__entries-container"
  />
</aside>
`;
